src/controllers/auth.controller.js

```diff
@@ -0,0 +1,325 @@
+  }
+
+  generateTokens(res, user);
+};
+
+const getUser = async (req, res) => {
+  const { userId } = req;
+
+  const user = await usersService.getOne({ id: userId });
+
+  res.json(usersService.normalize(user));
+};
+
+const resetPassword = async (req, res) => {
+  const { email } = req.body;
+
+  if (!email) {
+    throw ApiError.badRequest('email is required', {
+      email: 'email is required',
+    });
+  }
+
+  const user = await usersService.getOne({ email });
+
+  if (!user) {
+    throw ApiError.badRequest('This email is not registered', {
+      email: 'This email is not registered',
+    });
+  }
+
+  const resetPasswordToken = jwtService.signConfirmToken(
+    usersService.normalize(user),
+  );
+
+  await usersService.update({ email }, { resetPasswordToken });
+
+  await emailService.sendResetPasswordEmail(email, resetPasswordToken);
+
+  res.send({ message: 'Check your email for reset password' });
+};
+
+const changePassword = async (req, res) => {
+  const { resetPasswordToken } = req.params;
+  const { password, confirmPassword } = req.body;
+
+  const user = jwtService.verify(resetPasswordToken);
+
+  if (!user) {
+    await usersService.update(
+      { resetPasswordToken },
+      {
+        resetPasswordToken: null,
+      },
+    );
+
+    throw ApiError.badRequest('The provided token is no longer valid', {
+      token: 'The token has expired or is invalid',
+    });
+  }
+
+  if (password !== confirmPassword) {
+    throw ApiError.badRequest('Passwords are differents', {
+      password: 'Passwords are differents',
+    });
+  }
+
+  const currentSchema = dynamicSchema(Object.keys({ password }, userSchema));
+  const { error, value } = currentSchema.validate({ password });
+
+  if (error) {
+    throw ApiError.badRequest(error.message, { password: error.message });
+  }
+
+  const hashedPassword = await bcrypt.hash(value, 10);
+
+  await usersService.update(
+    { id: user.id },
+    {
+      resetPasswordToken: null,
+      password: hashedPassword,
+    },
+  );
+
+  res.send('Password is changed');
+};
+
+const updateUser = async (req, res) => {
+  const { userId } = req;
+  const { newPassword, confirmPassword, email, name, password } = req.body;
+
+  const user = await usersService.getOne({ id: userId });
+
+  const updatedFields = {};
+  const validateFields = {};
+
+  if (name) {
+    updatedFields.name = name;
+    validateFields.name = name;
+  }
+
+  let isPasswordValid = false;
+
+  if (password) {
+    isPasswordValid = await bcrypt.compare(password, user.password);
+  }
+
+  if (newPassword || confirmPassword) {
+    if (!password || !newPassword || !confirmPassword) {
+      throw ApiError.badRequest('All passwords are required', {
+        password: 'All passwords are required',
+      });
+    }
+
+    if (newPassword !== confirmPassword || !isPasswordValid) {
+      throw ApiError.badRequest('Passwords are differents', {
+        password: 'Passwords are differents',
+      });
```

Comment:
The error message 'Passwords are differents' contains a grammatical error. Consider changing it to 'Passwords are different'.